src/bootstrap/test.rs

```diff
@@ -846,12 +846,30 @@ fn check_if_browser_ui_test_is_installed_global(npm: &Path, global: bool) -> boo
         .output()
         .map(|output| String::from_utf8_lossy(&output.stdout).into_owned())
         .unwrap_or(String::new());
-    lines.contains(&" browser-ui-test@")
+    lines
+        .lines()
+        .filter_map(|l| l.split(" browser-ui-test@").skip(1).next().map(|v| v.to_owned()))
+        .next()
 }
 
-fn check_if_browser_ui_test_is_installed(npm: &Path) -> bool {
-    check_if_browser_ui_test_is_installed_global(npm, false)
-        || check_if_browser_ui_test_is_installed_global(npm, true)
+fn get_browser_ui_test_version(npm: &Path) -> Option<String> {
+    get_browser_ui_test_version_inner(npm, false)
+        .or_else(|| get_browser_ui_test_version_inner(npm, true))
+}
+
+fn compare_browser_ui_test_version(installed_version: String, src: &Path) {
+    match fs::read_to_string(src.join("src/ci/docker/scripts/browser-ui-test.version")) {
```

Comment:
I would trim at least the from-file version somewhere to avoid emitting errors if the file gains a newline character.

Reply:
Good point!